docs/design/jet/025-jet-job-member-pruning.md

```diff
@@ -0,0 +1,208 @@
+`ExecutionPlanBuilder` we checks that all vertices in DAG does not work without input and then we can apply member pruning.
+
+### Processor pruning
+
+For processor pruning we would like to separate two kinds of pruning: **intra-member** processor pruning, which
+is applicable only for processor creation elimination inside one member, and **inter-member** processor pruning.
+
+#### Use cases for processor pruning
+
+##### Processor pruning for Scan -> Transform -> Aggregate
+
+1. Single-staged aggregation.
+
+![Scan + Scan -> HashJoin](https://svgshare.com/i/tTM.svg)
+
+On the picture above we can see optimized example from member pruning case. We can go further and just don't create 
+processors which are not participating in data processing by tuning local parallelism parameter with 
+partition involvement knowledge: 
+
+![Scan + Scan -> HashJoin](https://svgshare.com/i/tUD.svg)
+
+So, it's a good target for **intra-member** processor pruning.
+
+2. Double-staged aggregation.
+```
+- COMBINE
+  - ACCUMULATE
+    - TRANSFORM
+      - SCAN[map, partitionKey=K]
+```
+![Scan + Scan -> HashJoin](https://jet-start.sh/docs/assets/arch-dag-4.svg)
+
+For 2-stage aggregation, honestly, an author don't see actual processor pruning abilities to optimize that case. 
+Potentially it may be done in another way - if we can translate 2-staged into single-staged on SQL opt phase and 
+then eliminate member(s) via member pruning.
+Even though, lets take a look on the picture above. Here, we can theoretically, eliminate scan and flatmap nodes. 
+In practice, processor logic is opaque to Jet and may have side effects which may break the job. 
+This example illustrates **inter-member** processor pruning.
+
+#### Solution design details
+
+##### Intra-member processor pruning
+To control processor creation and parallelism within one member, we would like to use various processor suppliers
+(`ProcessorMetaSupplier` or `ProcessorSupplier`). The correctness of this method will totally rely on how DAG constructor.
+Since Partition Pruning initiative was introduced to align PredicateAPI and SQL functionality and performance, we will
+rely on SQL optimizer input and DAG construction phase in `CreateDagVisitor`. Here we describe Jet API to support processor
+pruning :
+- create new `lazyForceTotalParallelismOne` (or override old `forceTotalParallelismOne`) PMS builder which does not 
+  cache member address to prevent wrong usage of cached plan;
+- new smart `_insert_this_name` PMS wrapper which computes preferred local parallelism based on input from SQL optimizer
+  and partitions owned by given member;
+
+##### Inter-member processor pruning
+
+After long discussions, we decided NOT support this kind of processor pruning, because it
+- relatively ineffective, because most of DAGs with broadcast edges were created with algorithm correctness in mind
+- hard to implement solution which fits most use cases;
+- doesn't fit the goal of general effort.
```

Comment:
```suggestion
After long discussions, we decided NOT to support this kind of processor pruning, because it
- is relatively ineffective since most DAGs with broadcast edges were created with algorithm correctness in mind,
- is hard to implement for most use cases, and
- doesn't fit the goal of general effort.
```